merino/providers/top_picks/provider.py

```
@@ -0,0 +1,86 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+"""Top Pick Navigational Queries Provider"""
+import logging
+from typing import Optional
+
+from merino.exceptions import BackendError
+from merino.providers.base import BaseProvider, BaseSuggestion, SuggestionRequest
+from merino.providers.top_picks.backends.protocol import TopPicksBackend, TopPicksData
+
+logger = logging.getLogger(__name__)
+
+
+class Suggestion(BaseSuggestion):
+    """Model for Top Pick Suggestion."""
+
+    block_id: int
+    is_top_pick: bool
+
+
+class Provider(BaseProvider):
+    """Top Pick Suggestion Provider."""
+
+    top_picks_data: TopPicksData
+
+    def __init__(
+        self,
+        backend: TopPicksBackend,
+        score: float,
+        name: str,
+        enabled_by_default: bool = False,
+    ) -> None:
+        self.backend = backend
+        self.score = score
+        self._name = name
+        self._enabled_by_default = enabled_by_default
+
+    async def initialize(self) -> None:
+        """Initialize the provider."""
+        try:
+            # Fetch Top Picks suggestions from domain list.
+            self.top_picks_data: TopPicksData = await self.backend.fetch()
+
+        except BackendError as backend_error:
+            logger.warning(
```

If a `BackendError` gets raised, this will leave `self.top_picks_data` undefined. We need to how to handle it.
- We can propagate the exception and let the application fail to start
- Or we can ignore it and retry `backend.fetch()` later, though we need to set an empty value for `this.top_picks_data`, otherwise, the subsequent calls on `query()` will fail

Since we're using a local file now, retrying is perhaps not gonna help that much. I think we can go with the first option. If the fetch fails, let's just crash the app.

Does that make sense?

---

Gotcha. Yeah, so shall I simply raise an exception (re raise) within the except block to crash the app? Conversely, I suppose having an empty list of suggestions instantiated,  should the backend data return noting seems decent too, we'd still get the log, but not crash.

---

Yeah, re-raise the exception and crash sounds reasonable in this case. If the JSON file is problematic, we should let it fail fast and hopefully we can capture it before it hits production. If this somehow happens in production, the crash is also fine as another merino instance will be spawned later.

---

Done @ncloudioj . Re -aises and passes context of `backend_error`, traceback very clear:

``` 
 File "/Users/taddeskorris/Documents/code/mozilla/merino-py/merino/providers/top_picks/provider.py", line 51, in initialize
    raise BackendError(
merino.exceptions.BackendError: Failed to fetch data from Top Picks Backend. Cannot decode file 'dev/top_picks.json'
```